Issue #2806011. Avoid calling drupal_goto() directly from rules actions.

xurizaemon opened the pull request:
See https://www.drupal.org/project/civicrm_entity/issues/2806011

xurizaemon committed:
Issue #2806011. Avoid calling drupal_goto() directly from rules actions.

jackrabbithanna commented on modules/civicrm_entity_actions/civicrm_entity_actions_action_schedule.inc:
```diff
@@ -41,26 +41,26 @@ function civicrm_entity_actions_action_schedule_action_info() {
 function civicrm_entity_actions_action_schedule_add_action($entity, $context = array()) {
   $base_url = str_replace('_', '-', $context['entity_type']); // civicrm-action_schedule
-  drupal_goto($base_url . '/add');
+  rules_action_drupal_goto($base_url . '/add');
```
If you use this function then this module will need to add a dependency to Rules in the .info file

jackrabbithanna replied:
alternatively you could add a if (module_exists('rules')) { new function, else drupal_goto

jackrabbithanna commented:
@xurizaemon I'm going to be cutting 7.x-2.0 stable next week. If you have time, and want to finish this off, we can get it in the 2.0 release

xurizaemon committed:
Issue #2806011. Implement our own Rules-preferring goto wrapper.

Preserve BC while preferring Rules shutdown method over Drupal's exit approach.

xurizaemon commented:
Sorry I'd missed yr earlier comments. Second approach seems sensible.

IDK if this is really a problem for anyone in the real world, so I'm OK with it not being merged!

I had five minutes to implement your suggestion but I don't have time / available test sites to test this with right now.

jackrabbithanna commented:
Your solution seems great for this in your second commit.  I'll test this and it'll go into 2.0. Thanks alot :)

xurizaemon commented:
![no-worries-alot](https://user-images.githubusercontent.com/105608/34084610-894b8bb2-e3e8-11e7-9164-d4d3c0943bdd.png)

jackrabbithanna commented:
Merging #129
